irohad/execution/impl/command_executor.cpp

```diff
@@ -0,0 +1,831 @@
+    }
+
+    auto account_asset_new = account_asset_builder.balance(new_balance)
+                             .accountId(command->accountId())
+                             .assetId(command->assetId())
+                             .build();
+    return makeExecutionResult(commands->upsertAccountAsset(account_asset_new),
+                               command_name);
+  }
+
+  ExecutionResult CommandExecutor::operator()(
+      const shared_model::detail::PolymorphicWrapper<shared_model::interface::AddPeer> &command) {
+    return makeExecutionResult(commands->insertPeer(command->peer()),
+                               "AddPeer");
+  }
+
+  ExecutionResult CommandExecutor::operator()(
+      const shared_model::detail::PolymorphicWrapper<shared_model::interface::AddSignatory> &command) {
+    std::string command_name = "AddSignatory";
+    auto result = commands->insertSignatory(command->pubkey()) | [&] {
+      return commands->insertAccountSignatory(command->accountId(),
+                                              command->pubkey());
+    };
+    return makeExecutionResult(result, command_name);
+  }
+
+  ExecutionResult CommandExecutor::operator()(
+      const shared_model::detail::PolymorphicWrapper<shared_model::interface::AppendRole> &command) {
+    return makeExecutionResult(
+        commands->insertAccountRole(command->accountId(), command->roleName()),
+        "AppendRole");
+  }
+
+  ExecutionResult CommandExecutor::operator()(
+      const shared_model::detail::PolymorphicWrapper<shared_model::interface::CreateAccount> &command) {
+    std::string command_name = "CreateAccount";
+    auto account =
+        account_builder
+            .accountId(command->accountName() + "@" + command->domainId())
+            .domainId(command->domainId())
+            .quorum(1)
+            .jsonData("{}")
+            .build();
+    auto domain = queries->getDomain(command->domainId());
+    if (not domain) {
+      return makeExecutionError(
+          (boost::format("Domain %s not found") % command->domainId()).str(),
+          command_name);
+    }
+    std::string domain_default_role = domain.value()->defaultRole();
+    // TODO: remove insert signatory from here ?
+    auto result = commands->insertSignatory(command->pubkey()) | [&] {
+      return commands->insertAccount(account);
+    } | [&] {
+      return commands->insertAccountSignatory(account.accountId(),
+                                              command->pubkey());
+    } | [&] {
+      return commands->insertAccountRole(account.accountId(),
+                                         domain_default_role);
+    };
+    return makeExecutionResult(result, "CreateAccount");
+  }
+
+  ExecutionResult CommandExecutor::operator()(
+      const shared_model::detail::PolymorphicWrapper<shared_model::interface::CreateAsset> &command) {
+    auto new_asset =
+        asset_builder.assetId(command->assetName() + "#" + command->domainId())
```

Comment:
check if "#" + commant->domainId() is needed

Reply:
It seems to be necessary as well